src/main/common/utils/numericUtils.ts

```diff
@@ -10,7 +10,19 @@ export function toNumberOrUndefined (value: any): number {
   }
 
   const strVal: string = value && value.toString().trim()
-  const numberVal: number = strVal.length > 0 ? _.toNumber(strVal) : undefined
 
-  return isNaN(numberVal) ? undefined : numberVal
+  // checks length, followed by a filter by not containing ',' or valid thousand separators
+  const numberVal: number = strVal.length > 0
+    ? (!strVal.includes(',') || containsAThousandSeparator(strVal))
+      ? numeral(strVal).value()
+      : strVal
+    : undefined
```

Comment:
Nested ternary expressions are a little hard to read, I think we could just use an `if` statement here.